### PR TITLE
[SPARK-9570] [DOCS] Consistent recommendation for submitting spark apps to YARN, -master yarn --deploy-mode x vs -master yarn-x'.

### PR DESCRIPTION
**Pull request by srowen**
Recommend `--master yarn --deploy-mode {cluster,client}` consistently in docs.
Follow-on to https://github.com/apache/spark/pull/8385
CC @nssalian

### CODE REVIEW

File: docs/running-on-yarn.md [hunk 1]
```diff
@@ -388,6 +390,6 @@ If you need a reference to the proper location to put log files in the YARN so t
...
-- In `yarn-cluster` mode, the local directories used by the Spark executors and the Spark driver will be the local directories configured for YARN (Hadoop YARN config `yarn.nodemanager.local-dirs`). If the user specifies `spark.local.dir`, it will be ignored. In `yarn-client` mode, the Spark executors will use the local directories configured for YARN while the Spark driver will use those defined in `spark.local.dir`. This is because the Spark driver does not run on the YARN cluster in `yarn-client` mode, only the Spark executors do.
+- In `cluster` mode, the local directories used by the Spark executors and the Spark driver will be the local directories configured for YARN (Hadoop YARN config `yarn.nodemanager.local-dirs`). If the user specifies `spark.local.dir`, it will be ignored. In `yarn-client` mode, the Spark executors will use the local directories configured for YARN while the Spark driver will use those defined in `spark.local.dir`. This is because the Spark driver does not run on the YARN cluster in `yarn-client` mode, only the Spark executors do.
```
Comment (tgravescs) [inline docs/running-on-yarn.md]: still using yarn-client  here.

File: docs/submitting-applications.md [hunk 2]
```diff
@@ -122,21 +123,23 @@ The master URL passed to Spark can be in one of the following formats:
...
 </td></tr>
-<tr><td> yarn-client </td><td> Connect to a <a href="running-on-yarn.html"> YARN </a> cluster in
-client mode. The cluster location will be found based on the HADOOP_CONF_DIR or YARN_CONF_DIR variable.
+<tr><td> <code>yarn</code> </td><td> Connect to a <a href="running-on-yarn.html"> YARN </a> cluster in
+        <code>client</code> or <code>cluster</code> mode depending on the value of <code>--deploy-mode</code>. 
+        The cluster location will be found based on the <code>HADOOP_CONF_DIR</code> or <code>YARN_CONF_DIR</code> variable.
 </td></tr>
-<tr><td> yarn-cluster </td><td> Connect to a <a href="running-on-yarn.html"> YARN </a> cluster in
-cluster mode. The cluster location will be found based on the HADOOP_CONF_DIR or YARN_CONF_DIR variable.
+<tr><td> <code>yarn-client</code> </td><td> Equivalent to <code>yarn</code> with <code>--deploy-mode client</code>
```
Comment (tgravescs) [inline docs/submitting-applications.md]: should we either leave these off or perhaps put deprecated by them?

Comment (nssalian) [inline docs/submitting-applications.md]: Shouldn't deploy-mode come first here?

Comment (srowen) [inline docs/submitting-applications.md]: Although deprecated feels a little strong ... yeah maybe best to sell it that way. I'd feel funny about completely omitting them as they have been used widely to date and still fully work.

Oops yes missed an occurrence of yarn-client, will fix

**Comment by tgravescs** [comment]
@srowen , what happened here, why was this closed, it looks like it was committed but I don't see anyone +1.  I was mostly good with it but it would have been nice to look at the final version.

**Comment by srowen** [comment]
Ah, I thought you had reviewed this and the previous iteration of this PR also? seemed pretty cut and dried to me at this stage. I don't know how long to wait for your explicit ack. In this case, being just docs, the downside of committing early is small -- you can always follow on with another commit if you really had something to add.

**Pull request closed without merging**